RFCs/FS-1128-better-interpolated-triple-quoted-strings.md

```diff
@@ -0,0 +1,119 @@
+
+Example:
+```
+let text = "3 curly braces delimit interpolation expression"
+$$$"""
+In a string literal with 3x$, { or {{ are simply treated as content
+whereas {{{text}}}.
+It is also allowed to have {, {{, } or }} adjacent to the interpolation delimiters as so:
+{{{{41+1}}}} = {42}
+"""
+```
+
+# Drawbacks
+
+Even though this only extends already existing syntax, it still can be considered yet another way of doing string interpolation. Moreover, it can't be fully aligned with analogous C# feature (raw string literals) and stay backward compatible at the same time, so this could be a source of confusion (but note that triple quoted string literals are already similar but not really aligned with C#'s raw string literals anyway).
+
+# Alternatives
+
+Alternatives:
+1. Try to fully align with C#'s raw string literals at the cost of backward compatiblity.
+2. Introduce alternative syntax for delimiting expression fills that does not clash with `{` and `}` which are common in JSON and certain other contexts.
+
+# Compatibility
+
+Please address all necessary compatibility questions:
+
+* Is this a breaking change?
+*No*
+* What happens when previous versions of the F# compiler encounter this design addition as source code?
+*A compiler error*
+* What happens when previous versions of the F# compiler encounter this design addition in compiled binaries?
+*AFAICT it should work, changes will hopefully be limited to lexing stage*
+* If this is a change or extension to FSharp.Core, what happens when previous versions of the F# compiler encounter this construct?
+*It is not a change/extension to FSharp.Core*
+
+# Pragmatics
+
+## Diagnostics
+
+* New compiler error for interpolated string literals with too many consecutive `{` or `}` characters.
+* Otherwise, same compiler errors as for regular (single-`$`) interpolated strings.
+
+## Tooling
+
+Compiler tooling should offer the same support for the extended form of interpolated string literals as it currently does for interpolated triple quoted literals. This includes colorization, autocompletion, and navigational features in an interpolated context.
+
+## Performance
+
+AFAICT there should be no impact on performance (although there might be if using certain constructs in regexes within .fsl files can have a big impact `XXX TODO`)
+
+## Scaling
+
+Please list the dimensions that describe the inputs for this new feature, e.g. "number of widgets" etc.  For each, estimate a reasonable upper bound for the expected size in human-written code and machine-generated code that the compiler will accept.
```

Comment:
I don’t understand “number of widgets” in this context. Also, why does this start with “please list…”? Is this a left over from the template?

Reply:
Yep, this part is left from a template, I only added a line at the bottom of this section - still need to figure this out